File: php/features.php
<?php
declare(strict_types=1);

// UnixConverter SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UnixConverterFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UnixConverterBaseFeature();
            case "test":
                return new UnixConverterTestFeature();
            default:
                return new UnixConverterBaseFeature();
        }
    }
}

File: py/features.py
# UnixConverter SDK feature factory

from feature.base_feature import UnixConverterBaseFeature
from feature.test_feature import UnixConverterTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UnixConverterBaseFeature(),
        "test": lambda: UnixConverterTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
